source/UIAUtils.py

```diff
@@ -286,18 +287,23 @@ def unregisterFromClientObject(self, element):
 			self.clientObject.RemovePropertyChangedEventHandler(element, handler)
 
 
-def _shouldUseUIAConsole(hwnd):
+def _shouldUseUIAConsole(hwnd: int) -> bool:
 	"""Determines whether to use UIA in the Windows Console."""
 	setting = config.conf['UIA']['winConsoleImplementation']
 	if setting == "UIA":
 		return True
-	# #7497: the UIA implementation in old conhost is incomplete, therefore we
-	# should ignore it.
-	return False
+	elif setting == "legacy":
+		return False
+	else:
+		# #7497: the UIA implementation in old conhost is incomplete, therefore we
+		# should ignore it.
+		# When the UIA implementation is improved, the below line will be replaced
+		# with a call to _shouldUseUIAConsole.

```

Comment:
Do you mean the following or is this a recursive call?
```suggestion
		# with a call to _isImprovedConhostTextRangeAvailable.
```